FAIMS/FAIMS3
# Improve UX for child record creation and management

# Feat: Improve UX for child record creation and management

## JIRA Ticket

[BSS-364](https://jira.csiro.com/browse/BSS-364)

## Description

BSS prefers a style for child records (buildings) where they are shown inline in the main form.  Current Fieldmark creates a link to the child record and shows the list of links instead.  Editing a child record jumps to a separate page for that form and saving comes back.

Fully embedding the child record form is probably too involved for v1 implementation but the experience of child records could be improved.

## Proposed Changes

- [x] Improve the layout of the RelatedRecordField to use the name of the child form (Building) in the action buttons
- [x] Derive the label on the add record button from the form name - instead of "Add child record" it could be "Add <FormName>" ("Add Building"). 
- [x] When child records exist, show a more detailed summary list, perhaps more like the updated record list in the notebook view with customisable fields shown
- [x] Update the 'publish' buttons on a child record to refer to the parent type 'Publish and return to Site' and 'Publish and New Building'
- [x] Use `RecordMetadata` in the child record list rather than `RecordLinkProps`, modify `listRecordMetadata` to return field values as well as metadata so we can include field values in the record listing

## How to Test

Update to the most recent Rapid Site Survey from github since it includes the summary fields configuration. 

Create a site record. Note the button to make new Buildings or add an existing building.   Create a new building, note the names on the buttons on that form.   Publish the building and return to the site, notice the fields shown in the building listing (currently 'building name' and 'building damage' but can be customised).  

Check that buildings are listed in the Site as expected. Add multiple buildings.

## Notes

This has exposed for me how awful the handling of records is in the app.  Too many similar types for records and usually the wrong one where you need to get some information like the type or field values.  Desperately need to refactor the interface to the records database to make it more orthogonal and less annoying.

## Checklist
